app/models/couch.rb

```diff
@@ -16,6 +16,10 @@ def self.search(params)
       .distinct
   end
 
+  def available_nights
+    self.nights.map { |night| night.date if night.reservation_id.nil? }
```

Comment:
@ski-climb @tmikeschu updated to self.nights.where(reservation_id: nil).pluck(:date) and everything looks good